shared/bots/repo_bots.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+from shared.bots.exceptions import (
+    OwnerWithoutValidBotError,
+    RepositoryWithoutValidBotError,
+)
+from shared.bots.github_apps import get_github_app_token
+from shared.bots.owner_bots import get_owner_or_appropriate_bot
+from shared.bots.public_bots import get_public_bot_token
+from shared.bots.types import TokenWithOwner
+from shared.config import get_config
+from shared.django_apps.codecov_auth.models import Owner, Service
+from shared.django_apps.core.models import Repository
+from shared.encryption.oauth import get_encryptor_from_configuration
+from shared.environment.environment import is_enterprise
+from shared.typings.torngit import GithubInstallationInfo
+
+encryptor = get_encryptor_from_configuration()
+
+log = logging.getLogger(__name__)
+
+
+def get_repo_particular_bot_token(repo) -> TokenWithOwner:
+    appropriate_bot = get_repo_appropriate_bot(repo)
+    token_dict = encryptor.decrypt_token(appropriate_bot.oauth_token)
+    token_dict["username"] = appropriate_bot.username
+    return token_dict, appropriate_bot
+
+
+def get_repo_appropriate_bot(repo: Repository) -> Owner:
+    if repo.bot is not None and repo.bot.oauth_token is not None:
+        log.info(
+            "Repo has specific bot",
+            extra=dict(repoid=repo.repoid, botid=repo.bot.ownerid),
+        )
+        return repo.bot
+    try:
+        return get_owner_or_appropriate_bot(repo.author)
```

Comment:
Changed from repo.owner to repo.author. Sqlalchemy has an `owner` foreign key to an Owner while django calls it `author`